src/main/java/org/mockito/internal/util/DefaultMockingDetails.java

```diff
@@ -36,17 +38,25 @@ public boolean isSpy(){
 
     @Override
     public Collection<Invocation> getInvocations() {
-        return getMockHandler(toInspect).getInvocationContainer().getInvocations();
+        return mockHandler().getInvocationContainer().getInvocations();
     }
 
     @Override
-    public Class<?> getMockedType() {
-        return getMockHandler(toInspect).getMockSettings().getTypeToMock();
+    public MockCreationSettings<?> getMockCreationSettings() {
+        return mockHandler().getMockSettings();
     }
 
-    @Override
-    public Set<Class<?>> getExtraInterfaces() {
-        return getMockHandler(toInspect).getMockSettings().getExtraInterfaces();
+    private InternalMockHandler<Object> mockHandler() {
+        assertGoodMock();
```

Comment:
I think the assertions should be on the constructor. This enable a fail fast behavior.